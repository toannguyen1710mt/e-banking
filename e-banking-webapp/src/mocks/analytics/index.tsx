export const ANALYTICS_DATA = [
  {
    title: 'Total Income',
    subtitle: 'Last 30 days income',
    isPositive: true,
    amount: 670000,
    percentageChange: 8,
  },
  {
    title: 'Total Expense',
    subtitle: 'Last 30 days income',
    isPositive: true,
    amount: 270000,
    percentageChange: 8,
  },
  {
    title: 'Total Investment',
    subtitle: 'Last 30 days income',
    isPositive: true,
    amount: 170000,
    percentageChange: 8,
  },
];
