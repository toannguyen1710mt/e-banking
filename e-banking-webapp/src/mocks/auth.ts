// Libs
import { Session } from 'next-auth';

// Interfaces
import { IUser } from '@/interfaces';

export const MOCK_DATA_USER: IUser = {
  accounts: [
    {
      id: 114,
      documentId: 'e5bavd05yrng3g850yazqj0q',
      name: 'OCB',
      accountNumber: '123456789208',
      balance: 3441800,
      type: 'Main',
      currency: '$',
    },
    {
      id: 115,
      documentId: 'kvelrmbr5yan4gyhz4wnoio3',
      name: 'ACB',
      accountNumber: '999332289888',
      balance: 84438800,
      type: 'Checkings',
      currency: '$',
    },
    {
      id: 116,
      documentId: 'nios4sn0bgw5u28nelo8fmao',
      name: 'VID',
      accountNumber: '106868929357',
      balance: 9235800,
      type: 'Savings',
      currency: '$',
    },
  ],
  avatar: 'https://i.ibb.co/SKHPQYq/avatar-default.webp',
  country: 'UK',
  phone: '123456789000',
  postal: '226',
  token:
    'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6ODIsImlhdCI6MTczNzA4NzMwNiwiZXhwIjoxNzM5Njc5MzA2fQ.6VPWVxS169pVk-4xBcC9TOj2e1DS10b_l3t9Grs5pU0',
  id: 82,
  documentId: 'ldrhsspx98g1ijlabnrclht8',
  username: 'phong',
  email: 'phong+1@gmail.com',
  provider: 'local',
  confirmed: true,
  blocked: false,
  announcements: true,
  updates: true,
  feedbacksAndSurvey: true,
  events: true,
  generalNotification: true,
  promotions: true,
  eventsNearMe: true,
  createdAt: '2025-01-15T04:54:45.295Z',
  updatedAt: '2025-01-15T04:54:45.623Z',
  publishedAt: '2025-01-15T04:54:45.296Z',
};

export const MOCK_DATA_USER_EMPTY_PROPS: IUser = {
  accounts: [
    {
      documentId: '11',
      accountNumber: '',
      balance: 0,
      currency: '$',
      id: 36,
      name: '',
      type: '',
    },
    {
      documentId: '12',
      accountNumber: '',
      balance: 0,
      currency: '$',
      id: 37,
      name: '',
      type: '',
    },
    {
      documentId: '12',
      accountNumber: '',
      balance: 0,
      currency: '$',
      id: 38,
      name: '',
      type: '',
    },
  ],
  avatar: 'https://i.ibb.co/SKHPQYq/avatar-default.webp',
  country: 'UK',
  email: 'thong+1@gmail.com',
  id: 43,
  phone: '123456789000',
  postal: '226',
  token: '123456',
  username: 'thong',
  documentId: '',
  provider: '',
  confirmed: false,
  blocked: false,
  createdAt: '',
  updatedAt: '',
  publishedAt: '',
};

export const MOCK_SESSION_DATA: Session = {
  user: {
    email: 'phong+1@gmail.com',
    id: 82,
    documentId: 'ldrhsspx98g1ijlabnrclht8',
    username: 'phong',
    provider: 'local',
    confirmed: true,
    blocked: false,
    postal: '344',
    phone: '324345346345',
    country: 'United States',
    avatar: 'https://i.ibb.co/SKHPQYq/avatar-default.webp',
    announcements: true,
    updates: true,
    feedbacksAndSurvey: true,
    events: true,
    generalNotification: true,
    promotions: true,
    eventsNearMe: true,
    createdAt: '2025-01-15T04:54:45.295Z',
    updatedAt: '2025-01-15T04:54:45.623Z',
    publishedAt: '2025-01-15T04:54:45.296Z',
    token:
      'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6ODIsImlhdCI6MTczNzA4NzMwNiwiZXhwIjoxNzM5Njc5MzA2fQ.6VPWVxS169pVk-4xBcC9TOj2e1DS10b_l3t9Grs5pU0',
  },
  expires: '2025-01-17T04:24:13.935Z',
};
