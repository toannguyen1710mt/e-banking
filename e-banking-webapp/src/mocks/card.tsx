export const ANALYTICS_CARD = [
  {
    title: 'Total Income',
    isPositive: true,
    subtitle: 'Last 30 days income',
  },
  {
    title: 'Total Expense',
    isPositive: true,
    subtitle: 'Last 30 days expenditure',
  },
  {
    title: 'Total Investment',
    subtitle: 'Last 30 days investment',
  },
];

export const MOCK_LIST_CARD = [
  {
    id: 1,
    documentId: 'zj103n8cbv857otlqpt3ks2h',
    holderName: 'donal trump',
    cardNumber: '123412341234',
    expireAt: '2026-05',
    ccv: '336',
    fullName: null,
    createdAt: '2024-12-18T02:37:12.625Z',
    updatedAt: '2024-12-18T02:37:12.625Z',
    publishedAt: '2024-12-18T02:37:12.623Z',
  },
  {
    id: 2,
    documentId: 'ny4mxu6br8v900c3an9p5u0f',
    holderName: 'Beck Kham',
    cardNumber: '106868929357',
    expireAt: '2028-03',
    ccv: '339',
    fullName: null,
    createdAt: '2024-12-18T04:27:04.764Z',
    updatedAt: '2024-12-18T04:27:04.764Z',
    publishedAt: '2024-12-18T04:27:04.762Z',
  },
  {
    id: 3,
    documentId: 'ljq5kv0f327w5sor2czgs07t',
    holderName: 'Hoang Tran',
    cardNumber: '123412341234',
    expireAt: '2025-07',
    ccv: '333',
    fullName: null,
    createdAt: '2024-12-18T18:24:58.801Z',
    updatedAt: '2024-12-18T18:24:58.801Z',
    publishedAt: '2024-12-18T18:24:58.799Z',
  },
  {
    id: 4,
    documentId: 's0bz93jb4tc5w7wc820o33c2',
    holderName: 'Test Holder Name',
    cardNumber: '123456789876',
    expireAt: '2026-02',
    ccv: '123',
    fullName: null,
    createdAt: '2024-12-19T01:39:29.501Z',
    updatedAt: '2024-12-19T01:39:29.501Z',
    publishedAt: '2024-12-19T01:39:29.498Z',
  },
];
