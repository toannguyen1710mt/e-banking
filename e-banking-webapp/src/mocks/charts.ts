export const MOCK_SPENDING_STATISTIC_CHART_DATA = {
  series: [
    {
      name: 'Data 1',
      data: [10, 41, 35, 51, 49, 91, 133],
    },
    {
      name: 'Data 2',
      data: [28, 69, 73, 102, 123, 142, 153],
    },
  ],
  categories: ['Mon', 'Tue', 'Wed', 'Thur', 'Fri', 'Sat', 'Sun'],
};

export const MOCK_BALANCE_STATISTICS_CHART_DATA = {
  series: [
    {
      name: 'Data 1',
      data: [98, 124, 166, 350, 40, 289, 96],
    },
    {
      name: 'Data 2',
      data: [80, 132, 169, 270, 49, 174, 90],
    },
  ],
  categories: ['Mon', 'Tue', 'Wed', 'Thur', 'Fri', 'Sat', 'Sun'],
};

export const MOCK_SERIES_EXPENSE_ANALYSIS = [22, 45, 35, 40];

export const MASTERCARD_CHART_MOCK = [65, 20, 15];
