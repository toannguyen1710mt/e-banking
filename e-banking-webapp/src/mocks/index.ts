export * from './options';
export * from './card';
export * from './charts';
export * from './transaction';
export * from './table';
export * from './transferConfirmData';
export * from './target';
export * from './analytics';
export * from './auth';
export * from './ServiceCard';
export * from './formMocks';
