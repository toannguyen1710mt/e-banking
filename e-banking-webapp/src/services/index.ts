export * from './http-client';
export * from './account';
export * from './transaction';
export * from './users';
