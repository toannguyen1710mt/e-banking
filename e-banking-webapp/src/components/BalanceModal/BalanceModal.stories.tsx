// Libs
import type { Meta, StoryObj } from '@storybook/react';
import { useEffect, useState } from 'react';

// Mocks data
import { MOCK_SESSION_DATA } from '@/mocks';

// Components
import BalanceModal from '.';

const meta = {
  title: 'Components/BalanceModal',
  component: BalanceModal,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  args: {},
} satisfies Meta<typeof BalanceModal>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    isOpen: true,
    onClose: () => {},
    session: MOCK_SESSION_DATA,
  },
  render: (args) => {
    const [isOpen, setIsOpen] = useState(args.isOpen);

    useEffect(() => {
      setIsOpen(args.isOpen);
    }, [args.isOpen]);

    const handleClose = () => {
      setIsOpen(false);
      if (args.onClose) {
        args.onClose();
      }
    };

    return <BalanceModal {...args} isOpen={isOpen} onClose={handleClose} />;
  },
};
