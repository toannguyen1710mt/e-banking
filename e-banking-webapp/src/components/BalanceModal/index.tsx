'use client';

import { useCallback, useEffect, useMemo, useState } from 'react';
import { CircularProgress } from '@nextui-org/react';
import { Session } from 'next-auth';

// Constants
import { createExpenseAnalysisOptions } from '@/constants';

// Interfaces
import { IAccount, ICard, TEXT_SIZE, TEXT_VARIANT } from '@/interfaces';

// Utils
import { formatNumberWithCommas } from '@/utils';

// Services
import { getAccountsByUserId, getListCardByAccountId } from '@/services';

// Mocks
import { MOCK_SERIES_EXPENSE_ANALYSIS } from '@/mocks';

// Components
import { Modal, Text } from '../common';
import { MyCards } from '../MyCards';
import { ExpenseAnalysis } from '../ExpenseAnalysis';

interface BalanceModalProps {
  session: Session;
  isOpen?: boolean;
  totalInvestment?: string;
  totalIncome?: string;
  onClose: () => void;
}

const BalanceModal = ({
  session,
  totalInvestment = '170,000',
  totalIncome = '670,000',
  isOpen = false,
  onClose,
}: BalanceModalProps) => {
  const [accounts, setAccounts] = useState<IAccount[]>([]);
  const [cardData, setCardData] = useState<ICard>({} as ICard);
  const [selectedAccount, setSelectedAccount] = useState<IAccount>(
    {} as IAccount,
  );

  useEffect(() => {
    const fetchData = async () => {
      try {
        const result = await getAccountsByUserId(session.user.id);

        if (result?.length) {
          setAccounts(result);
          setSelectedAccount(result[0]);
          const documentId = result[0]?.documentId;

          if (documentId) {
            const { cards } = await getListCardByAccountId(
              documentId as string,
            );

            setCardData(cards[0] || {});
          }
        }
      } catch (error) {
        console.error('Error fetching data:', error);
      }
    };

    fetchData();
  }, [session?.user?.accounts, session?.user?.id]);

  const handleCardSelect = useCallback((account: IAccount) => {
    setSelectedAccount(account);
  }, []);

  const totalBalance = useMemo(
    () => accounts?.reduce((total, account) => total + account.balance, 0),
    [accounts],
  );

  return (
    <Modal
      isOpen={isOpen}
      onClose={onClose}
      placement='center'
      classNames={{
        base: 'max-w-xs lg:max-w-4xl p-0 rounded-xl',
        closeButton: 'z-50',
      }}
    >
      <div className='absolute left-4 top-4'>
        <Text as='h4' className='text-base font-medium text-primary-200'>
          Good Evening,{' '}
          <Text
            as='span'
            variant={TEXT_VARIANT.DEFAULT}
            className='font-normal'
          >
            {session?.user?.username || ''}
          </Text>
        </Text>
      </div>
      <div className='flex flex-col rounded-xl bg-lightGraySolid lg:flex-row'>
        <div className='flex flex-1 flex-col rounded-xl bg-white pb-[48px] pl-3 pr-[23px] pt-[56px]'>
          <div className='flex flex-col items-start justify-between gap-2 md:flex-row'>
            <div className='flex flex-col gap-2'>
              <Text
                as='h5'
                className='text-sm font-semibold'
                variant={TEXT_VARIANT.DEFAULT}
              >
                Current Balance
              </Text>
              <Text
                as='span'
                className='font-extrabold'
                variant={TEXT_VARIANT.DEFAULT}
              >
                ${formatNumberWithCommas(selectedAccount?.balance || 0)}
                <Text
                  as='span'
                  size={TEXT_SIZE['XS']}
                  variant={TEXT_VARIANT.DEFAULT}
                  className='font-semibold opacity-50'
                >
                  /${totalIncome}
                </Text>
              </Text>
            </div>
            <div className='flex flex-col gap-2'>
              <Text
                as='h5'
                className='text-sm font-semibold'
                variant={TEXT_VARIANT.DEFAULT}
              >
                Total Investments
              </Text>
              <Text
                as='span'
                className='font-extrabold'
                variant={TEXT_VARIANT.DEFAULT}
              >
                ${totalInvestment}
              </Text>
            </div>
          </div>

          <div className='relative mx-auto mt-9 flex h-[100px] w-[100px] items-center justify-center'>
            <CircularProgress
              classNames={{
                svg: 'w-full h-full drop-shadow-md',
                indicator: 'stroke-[#2A9D8F]',
                track: 'stroke-[#2A9D8F]/10',
                value: 'text-medium font-extrabold text-black',
              }}
              showValueLabel={false}
              strokeWidth={3}
              value={70}
            />
            <div className='absolute flex flex-col items-center'>
              <Text
                className='text-center text-sm font-medium'
                variant={TEXT_VARIANT.DEFAULT}
              >
                Spend
              </Text>
              <Text
                as='span'
                className='text-center font-extrabold'
                variant={TEXT_VARIANT.DEFAULT}
              >
                70%
              </Text>
            </div>
          </div>
        </div>

        {/* My Cards and Expense Analysis */}
        <div className='flex flex-1 flex-col py-8 pl-4 pr-3'>
          <Text as='h4' variant={TEXT_VARIANT.DEFAULT}>
            My Cards
          </Text>
          <div className='my-4'>
            <MyCards
              expireDate={cardData?.expireAt}
              accounts={accounts as IAccount[]}
              onCardSelect={handleCardSelect}
            />
          </div>
          <Text
            size={TEXT_SIZE['XS']}
            variant={TEXT_VARIANT.DEFAULT}
            className='max-w-[180px] font-normal'
          >
            You have a total of{' '}
            <Text
              as='span'
              size={TEXT_SIZE['XS']}
              variant={TEXT_VARIANT.DEFAULT}
              className='font-bold'
            >
              ${formatNumberWithCommas(totalBalance || 0)}
            </Text>{' '}
            available in your wallet
          </Text>

          <div className='mt-2.5'>
            <ExpenseAnalysis
              options={createExpenseAnalysisOptions(
                formatNumberWithCommas(totalBalance || 0),
              )}
              series={MOCK_SERIES_EXPENSE_ANALYSIS}
            />
          </div>
        </div>
      </div>
    </Modal>
  );
};

export default BalanceModal;
