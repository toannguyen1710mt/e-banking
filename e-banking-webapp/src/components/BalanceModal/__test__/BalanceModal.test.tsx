import { fireEvent, render, waitFor } from '@testing-library/react';

// Services
import * as services from '@/services';

// Mocks
import { MOCK_DATA_USER, MOCK_SESSION_DATA } from '@/mocks';

// Component
import BalanceModal from '..';

jest.mock('react-apexcharts', () => {
  return {
    __esModule: true,
    default: () => <div>Mock ApexCharts</div>,
  };
});

jest.mock('@/services', () => ({
  getAccountsByUserId: jest.fn(),
}));

describe('BalanceModal component', () => {
  beforeEach(() => {
    (services.getAccountsByUserId as jest.Mock).mockResolvedValue(
      MOCK_DATA_USER.accounts,
    );
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('Should render snapshot correctly', () => {
    expect(
      render(
        <BalanceModal
          session={MOCK_SESSION_DATA}
          onClose={jest.fn()}
          isOpen={true}
        />,
      ),
    ).toMatchSnapshot();
  });

  it('Should render snapshot when is open false', () => {
    const { asFragment } = render(
      <BalanceModal session={MOCK_SESSION_DATA} onClose={jest.fn()} />,
    );

    expect(asFragment).toMatchSnapshot();
  });

  it('Should handle select card.', async () => {
    const { getByLabelText } = render(
      <BalanceModal
        session={MOCK_SESSION_DATA}
        onClose={jest.fn()}
        isOpen={true}
      />,
    );

    await waitFor(async () => {
      fireEvent.click(getByLabelText('card-item-1'));
    });

    expect(getByLabelText('card-item-1')).toBeInTheDocument();
  });

  it('Should handle close modal.', async () => {
    const mockOnClose = jest.fn();

    const { getByLabelText } = render(
      <BalanceModal
        session={MOCK_SESSION_DATA}
        onClose={mockOnClose}
        isOpen={true}
      />,
    );

    await waitFor(async () => {
      fireEvent.click(getByLabelText('Close'));
    });

    expect(mockOnClose).toHaveBeenCalled();
  });
});
