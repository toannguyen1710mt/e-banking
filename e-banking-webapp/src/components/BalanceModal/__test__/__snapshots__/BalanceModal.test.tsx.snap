// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BalanceModal component Should render snapshot correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div
      aria-hidden="true"
    />
    <span
      aria-hidden="true"
      data-focus-scope-start="true"
      hidden=""
    />
    <div
      tabindex="-1"
    >
      <div
        aria-hidden="true"
        class="z-50 bg-overlay/50 backdrop-opacity-disabled w-screen h-screen fixed inset-0"
        style="opacity: 0;"
      />
      <div
        class="flex w-screen fixed inset-0 z-50 overflow-x-auto justify-center h-[--visual-viewport-height] items-start sm:items-start overflow-y-auto [--scale-enter:100%] [--scale-exit:100%] [--slide-enter:0px] [--slide-exit:80px] sm:[--scale-enter:100%] sm:[--scale-exit:103%] sm:[--slide-enter:0px] sm:[--slide-exit:0px]"
        data-slot="wrapper"
        style="--visual-viewport-height: 768px; opacity: 0; transform: translateY(var(--slide-exit)) scale(var(--scale-exit));"
      >
        <section
          aria-modal="true"
          class="flex flex-col relative z-50 w-full box-border bg-content1 outline-none mx-1 sm:mx-6 sm:my-16 shadow-small my-16 !my-auto max-w-xs lg:max-w-4xl p-0 rounded-xl"
          data-dismissable="true"
          data-open="true"
          id=":r0:"
          role="dialog"
          tabindex="-1"
        >
          <div
            style="border: 0px; clipPath: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
          >
            <button
              aria-label="Dismiss"
              id="react-aria-:r5:"
              style="width: 1px; height: 1px;"
              tabindex="-1"
            />
          </div>
          <button
            aria-label="Close"
            class="group inline-flex items-center box-border whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] h-12 gap-3 [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none data-[hover=true]:opacity-hover border-transparent max-h-10 text-sm leading-[20px] absolute appearance-none select-none top-1 right-1 rtl:left-1 rtl:right-[unset] p-2 rounded-full active:bg-default-200 tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 bg-transparent max-h-none min-w-[20px] w-[20x] text-foreground-200 justify-end hover:bg-transparent data-[focus=true]:bg-transparent z-50"
            role="button"
            type="button"
          >
            <svg
              fill="currentColor"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19 3H5C3.89543 3 3 3.89543 3 5V19C3 20.1046 3.89543 21 5 21H19C20.1046 21 21 20.1046 21 19V5C21 3.89543 20.1046 3 19 3Z"
                stroke="#264653"
                strokeLinecap="round"
                 strokeLinejoin="round"
                strokeWidth="2"
              />
              <path
                d="M9 9L15 15"
                stroke="#264653"
                strokeLinecap="round"
                 strokeLinejoin="round"
                strokeWidth="2"
              />
              <path
                d="M15 9L9 15"
                stroke="#264653"
                strokeLinecap="round"
                 strokeLinejoin="round"
                strokeWidth="2"
              />
            </svg>
          </button>
          <div
            class="absolute left-4 top-4"
          >
            <h4
              class="text-primary-200 text-base font-medium text-primary-200"
            >
              Good Evening,
               
              <span
                class="text-foreground-100 text-base font-normal"
              >
                phong
              </span>
            </h4>
          </div>
          <div
            class="flex flex-col rounded-xl bg-lightGraySolid lg:flex-row"
          >
            <div
              class="flex flex-1 flex-col rounded-xl bg-white pb-[48px] pl-3 pr-[23px] pt-[56px]"
            >
              <div
                class="flex flex-col items-start justify-between gap-2 md:flex-row"
              >
                <div
                  class="flex flex-col gap-2"
                >
                  <h5
                    class="text-foreground-100 text-sm font-semibold"
                  >
                    Current Balance
                  </h5>
                  <span
                    class="text-foreground-100 text-base font-extrabold"
                  >
                    $
                    0
                    <span
                      class="text-foreground-100 text-xs font-semibold opacity-50"
                    >
                      /$
                      670,000
                    </span>
                  </span>
                </div>
                <div
                  class="flex flex-col gap-2"
                >
                  <h5
                    class="text-foreground-100 text-sm font-semibold"
                  >
                    Total Investments
                  </h5>
                  <span
                    class="text-foreground-100 text-base font-extrabold"
                  >
                    $
                    170,000
                  </span>
                </div>
              </div>
              <div
                class="relative mx-auto mt-9 flex h-[100px] w-[100px] items-center justify-center"
              >
                <div
                  aria-valuemax="100"
                  aria-valuemin="0"
                  aria-valuenow="70"
                  aria-valuetext="70%"
                  class="flex flex-col justify-center gap-1 max-w-fit items-center"
                  id="react-aria-:r6:"
                  role="progressbar"
                >
                  <div
                    class="relative block"
                  >
                    <svg
                      class="z-0 relative overflow-hidden text-primary w-full h-full drop-shadow-md"
                      fill="none"
                      strokeWidth="3"
                      viewBox="0 0 32 32"
                    >
                      <circle
                        class="h-full stroke-[#2A9D8F]/10"
                        cx="16"
                        cy="16"
                        r="13"
                        role="presentation"
                        stroke-dasharray="81.68140899333463 81.68140899333463"
                        stroke-dashoffset="0"
                        strokeLinecap="round"
                        transform="rotate(-90 16 16)"
                      />
                      <circle
                        class="h-full transition-all !duration-500 stroke-[#2A9D8F]"
                        cx="16"
                        cy="16"
                        r="13"
                        role="presentation"
                        stroke-dasharray="81.68140899333463 81.68140899333463"
                        stroke-dashoffset="81.68140899333463"
                        strokeLinecap="round"
                        transform="rotate(-90 16 16)"
                      />
                    </svg>
                  </div>
                </div>
                <div
                  class="absolute flex flex-col items-center"
                >
                  <p
                    class="text-foreground-100 text-center text-sm font-medium"
                  >
                    Spend
                  </p>
                  <span
                    class="text-foreground-100 text-base text-center font-extrabold"
                  >
                    70%
                  </span>
                </div>
              </div>
            </div>
            <div
              class="flex flex-1 flex-col py-8 pl-4 pr-3"
            >
              <h4
                class="text-foreground-100 text-base font-medium"
              >
                My Cards
              </h4>
              <div
                class="my-4"
              >
                <div
                  class="flex flex-wrap gap-4"
                />
              </div>
              <p
                class="text-foreground-100 text-xs max-w-[180px] font-normal"
              >
                You have a total of
                 
                <span
                  class="text-foreground-100 text-xs font-bold"
                >
                  $
                  0
                </span>
                 
                available in your wallet
              </p>
              <div
                class="mt-2.5"
              >
                <div
                  class="flex flex-col relative overflow-hidden h-auto text-foreground box-border outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium rounded-large transition-transform-background motion-reduce:transition-none max-w-[250px] bg-background-300 p-[10px]"
                  tabindex="-1"
                >
                  <div
                    class="flex z-10 w-full justify-start items-center shrink-0 overflow-inherit color-inherit subpixel-antialiased rounded-t-large p-0"
                  >
                    <p
                      class="text-foreground-200 text-base font-medium"
                    >
                      Expense Analysis
                    </p>
                  </div>
                  <div
                    class="mt-2 flex gap-4"
                  >
                    <div
                      class="grid grid-cols-2 content-center gap-x-5 gap-y-3"
                    >
                      <div
                        class="flex h-3 items-center gap-[5px]"
                      >
                        <div
                          class="h-[8px] w-[8px] rounded-full bg-[#8AB17D]"
                        />
                        <p
                          class="text-primary-200 font-medium text-2xs text-white opacity-50"
                        >
                          Amazon
                        </p>
                      </div>
                      <div
                        class="flex h-3 items-center gap-[5px]"
                      >
                        <div
                          class="h-[8px] w-[8px] rounded-full bg-[#2A9D8F]"
                        />
                        <p
                          class="text-primary-200 font-medium text-2xs text-white opacity-50"
                        >
                          Netflix
                        </p>
                      </div>
                      <div
                        class="flex h-3 items-center gap-[5px]"
                      >
                        <div
                          class="h-[8px] w-[8px] rounded-full bg-[#F4A261]"
                        />
                        <p
                          class="text-primary-200 font-medium text-2xs text-white opacity-50"
                        >
                          Apple
                        </p>
                      </div>
                      <div
                        class="flex h-3 items-center gap-[5px]"
                      >
                        <div
                          class="h-[8px] w-[8px] rounded-full bg-[#E9C46A]"
                        />
                        <p
                          class="text-primary-200 font-medium text-2xs text-white opacity-50"
                        >
                          Figma
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            style="border: 0px; clipPath: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
          >
            <button
              aria-label="Dismiss"
              id="react-aria-:r9:"
              style="width: 1px; height: 1px;"
              tabindex="-1"
            />
          </div>
        </section>
      </div>
    </div>
    <span
      aria-hidden="true"
      data-focus-scope-end="true"
      hidden=""
    />
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`BalanceModal component Should render snapshot when is open false 1`] = `[Function]`;
